app/grants/views.py

```diff
@@ -686,7 +689,10 @@ def get_grants_by_filters(
     three_months_ago = timezone.now() - timezone.timedelta(days=90)
 
     # 1. Filter grants by network and hidden = false
-    _grants = Grant.objects.filter(network=network, hidden=False)
+    _grants = Grant.objects.filter(network=network)
+    
+    if not hidden_grants:
+        _grants = Grant.objects.filter(hidden=False)
```

Comment:
shouldn't this be

```
    if not hidden_grants:
        _grants = _grants.filter(hidden=False)
 ```
 
 to keep the network filter + hit db once

Reply:
Thanks for catching this @chibie!